src/openrct2/editor.c

```diff
@@ -528,3 +529,174 @@ bool editor_check_park()
 
 	return true;
 }
+
+void game_command_edit_scenario_options(sint32* eax, sint32* ebx, sint32* ecx, sint32* edx, sint32* esi, sint32* edi, sint32* ebp)
+{
+	if (*ebx & GAME_COMMAND_FLAG_APPLY) {
+		switch (*ecx) {
+		case EDIT_SCENARIOOPTIONS_SETNOMONEY:
+			if (gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) {
+				if (*edx != 0) {
```

Comment:
To prevent code duplication move the if statement to the different code rather than repeat the same code.